spinn_utilities/make_tools/file_convertor.py

```diff
@@ -0,0 +1,453 @@
+import os
+import re
+import sys
+
+TOKEN = chr(30)  # Record Sperator
+
+STRING_REGEXP = re.compile('"([^"]|\\"|(""))*"')
+FORMATEXP = re.compile("%\d*(?:\.\d+)?[cdiksuxR]")
+LOG_END_REGEX = re.compile('\)(\s)*;')
+END_COMMENT_REGEX = re.compile("/*/")
+LOG_START_REGEX = re.compile(
+    "log_((info)|(error)|(debug)|(warning))(\s)*\(")
+
+# Status values
+NORMAL_CODE = 0
+COMMENT = NORMAL_CODE + 1
+IN_LOG = COMMENT + 1
+IN_LOG_CLOSE_BRACKET = IN_LOG + 1
+
+MINIS = {"log_info(": "log_mini_info(",
+         "log_error(": "log_mini_error(",
+         "log_debug(": "log_mini_debug(",
+         "log_warning(": "log_mini_warning("}
+
+LEVELS = {"log_info(": "[INFO]",
+          "log_error(": "[ERROR]",
+          "log_debug(": "[DEBUG]",
+          "log_warning(": "[WARNING]"}
+
+MAX_LOG_PER_FILE = 100
+
+
+class FileConvertor(object):
+    # __slots__ = [
+    #    "_dest", "_dest_basename", "_src", "_src_basename"]
+
+    # REGEXP = re.compile("%(?:\d+\$)?[dfsu]")
+
+    def __init__(self, src, dest, dict, range_start):
+        self._src = os.path.abspath(src)
+        self._dest = os.path.abspath(dest)
+        self._message_id = range_start
+        self._dict = dict
+
+    def run(self):
+        if not os.path.exists(self._src):
+            raise Exception("Unable to locate source {}".format(src))
+        dest_dir = os.path.dirname(os.path.realpath(self._dest))
+        if not os.path.exists(dest_dir):
+            os.makedirs(dest_dir)
+        self.convert_c()
+        return self._message_id
+
+    def convert_c(self):
+        with open(self._src) as src_f:
+            with open(self._dest, 'w') as dest_f:
+                dest_f.write(
+                    "// DO NOT EDIT! THIS FILE WAS GENERATED FROM {}\n\n"
+                    .format(self.unique_src()))
+                self._too_many_lines = 2
+                self._status = NORMAL_CODE
+                for self._line_num, self._text in enumerate(src_f):
+                    if self._too_many_lines > 0:
```

Comment:
self._too_many_lines not created during constructor, nor in slots

Reply:
same for _status, _line_num, _test, _comment_start,

Reply:
added to slots. CRITICAL!

 Did not add to init as I actually prefer the never created error over the null pointer error.

Reply:
phft to you. ill bring in rowley tomorrow and we can have a mediator on whom is right

Reply:
A quick check shows that too_many_lines is set on line 60 then used in the next few lines within the same function.  There is one reference outside the function where this variable is added to - however this addition seems to then be ignored... this seems incorrect; the variable can probably either be local or should be dealt with correctly later.

Reply:
Looks like _line_num can also be passed down?

Reply:
_comment_start looks to be write-only

Reply:
too_many_lines is added to in 
_write_log_method which is called from 
_process_line_in_log (and others) which is called from
_process_line (and others) which is called from 
inside the for loop which does skip lines in based on -_too_many_lines